docs/root/configuration/http/http_filters/thrift_to_metadata_filter.rst

```diff
@@ -0,0 +1,49 @@
+.. _config_http_filters_thrift_to_metadata:
+
+Envoy Thrift-To-Metadata Filter
```

Comment:
Can you explain the relationship between this and the existing Thrift proxy support in Envoy? When does someone use one vs. the other? @zuercher

Reply:
Sure. 

First of all, `thrift_proxy` is a network filter, and `thrift_to_metadata` is an HTTP filter.

`thrift_proxy` supports thrift RPC in HTTP 1.1 model, i.e., send a thrift serialized request over TCP and wait for the response. It's required to receive full response before send the next request. An example is [Apache THeader Transport](https://github.com/apache/thrift/blob/master/lib/py/src/transport/THeaderTransport.py)

`thrift_to_metadata` supports thrift serialized payload over HTTP, i.e., put RPC bytes in the POST body. Therefore, we could exploit all existing HTTP features and HTTP envoy filter. Moreover, we could utilize HTTP2 or HTTP3, which brings multiplexing and HOL avoidance. An example is [Apache THTTP](https://github.com/apache/thrift/blob/master/lib/py/src/transport/THttpClient.py)

While we use envoy HCM to proxy THTTP, it's natural to have a HTTP filter to extract thrift metadata, e.g., method name, as we do in `thrift_proxy`, which motivates us to implement this filter.

Moreover, we want to support the feature `payload_to_metadata` to extract thrift body information for other filters usage, but it won't be implemented at the first version so I mark it wip in the proto.

Reply:
OK, makes sense. @zuercher can you do primary review on this?

Reply:
done here
https://github.com/envoyproxy/envoy/pull/33919#pullrequestreview-2072613349